### ISSUE TITLE
Docs: link to "blanket implementation"

### ISSUE THREAD

**Issue by flip111** [rust-lang/rust#52551]
Hello, i'm new to rust but i had some problem finding relating documentation.

I started with this method here https://doc.rust-lang.org/std/primitive.char.html#method.to_lowercase

I see `to_lowercase` returns a `ToLowercase` struct, after that the `to_string()` function was called. When i click on the documentation for `ToLowercase` here https://doc.rust-lang.org/std/char/struct.ToLowercase.html i see nothing about this `to_string()` function. After asking around i've heard it's provided by `Display` somehow. But also clicking Display on the last page, which goes to https://doc.rust-lang.org/std/fmt/trait.Display.html has no mention of this to_string() method.

From what i've heard this is a "blanket implementation". Can those be added to the docs? So i can follow a logical path that match with the code? Maybe it will add more fluff .. but i think there is an easy solution to that: just hide it behind some button how the `[-]` buttons work for descriptions.

**Comment by hkcorac** [comment]
I guess your `c` is a `char`. If so you might want to check the docs of `to_lowercase()` again. It returns an iterator or one or more characters (I'm just copying/pasting the docs). If you're sure that the input is an ascii character, you can use `to_ascii_lowercase()` function instead.

Example:

```rust
ret.push_str(c.to_ascii_lowercase().to_string());
```

Or, an iterator of chars can be collect into a string like this:

```rust
ret.push_str(c.to_lowercase().collect::<String>());
```

Docs: <https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect>

By the way, I'm brand new to Rust too, haha. I don't understand your error message much. Hopefully I can help with above.

**Comment by flip111** [comment]
@hkcorac you are right. However my issue is about that i can't find a relationship between `to_lowercase()` -> `ToLowercase` -> `to_string()` by following the docs.

**Comment by hkcorac** [comment]
@flip111 

I'm still not sure about the error message that your Rust compiler produced. But I understand your concern. Actually it took me awhile with that too (before when I was just starting, not because of this issue).

The `to_string()` function belongs to this trait: <https://doc.rust-lang.org/std/string/trait.ToString.html> -- when you scroll down to the _Implementors_ section, you can see this line:

```rust
impl<T> ToString for T where T: Display + ?Sized;
```

I learned about `ToString` when reading one of the compiler error messages.

My Rust `1.27.2` compiles this:

```rust
fn main() {
    println!("{}", 'a'.to_lowercase().to_string());
}
```

Maybe you could update to latest Rust version?

**Comment by flip111** [comment]
I'm going to remove that error message from my first point. It's just adding noise to the discussion. I was on an old compiler version yes.

So the path (url links) `to_lowercase()` -> `ToLowercase` -> `Display` is there. But `Display` does not link to `ToString`

**Comment by aa5799a8-35c3-47c6-8a63-8faf08caff50** [comment]
@flip111 

Now I completely understand your original question. Sorry I didn't read it carefully.

I agree that the original error message led to `Display`'s documentation, but that documentation doesn't mention `ToString` trait.

From my view, I think the compiler could have produced better message. `Display`'s documentation doesn't need to  mention `ToString`, since both are not related.

My experience is: searching for `to_string` could help find out the original trait name. From there I can check its implementors. In many cases, the top bar search box could provide better results than external search engines. Maybe it could help you next time...

**Comment by hkcorac** [comment]
@flip111 

Now I completely understand your original question. Sorry I didn't read it carefully.

I agree that the original error message led to Display's documentation, but that documentation doesn't mention ToString trait.

From my view, I think the compiler could have produced better message. Display's documentation doesn't need to mention ToString, since both are not related.

My experience is: searching for to_string could help find out the original trait name. From there I can check its implementors. In many cases, the top bar search box could provide better results than external search engines. Maybe it could help you next time...

(Sorry, I was replying via a temporary account, ha-ha you caught me)  :-D

**Comment by flip111** [comment]
Searching for `to_string` works because i know what i am looking for. In other cases i just like to explore new information, that's why i propose the documentation also includes "blanket implementation" (if this is the right term here for this thing)

**Comment by flip111** [comment]
Blanket documentation now shows up, although it would still be nice to get a list of all the methods that can be called.
